### ISSUE TITLE
Bug: CV measurement fails silently

### ISSUE THREAD

**Issue by ilka-schulz** [Zahner-elektrik/Thales-Remote-Python#17]
I figured that CV measurements fail silently.

# Steps to reproduce
I started a CV measurement just like in the [example](https://github.com/Zahner-elektrik/Thales-Remote-Python/blob/main/Examples/CyclicVoltammetry/CyclicVoltammetry.ipynb).

Btw, I set the output path to:

```python
device: ThalesRemoteScriptWrapper = ...
device.setCVOutputPath("c:\\THALES")
```

The check function does not throw an exception:
```python
device.checkCVSetup()
```

# Expected behavior

Calling
```python
device.measureCV()
```
should run a CV measurement.

# Actual Behavior

There is no indication on the remote python application that anything is going wrong except that the measurement does not terminate after the expected duration.

However, the local Term software spams the console with this repeated error message:
```text
DIR SET c:\thales
ERROR ILLEGAL CMD
```
btw: I did not specify `c:\thales` as the output directory but `c:\THALES`. This would potentially later break toolchains, e.g. file transfer to the remote client as well as any further data processing.

**Comment by ilka-schulz** [comment]
The failure apparently is not silent, I accidentally handled the exception somewhere (sorry, I am not used to using exceptions for error handling as I mostly write Rust...)

This is the traceback:
```text
File "/home/user/.local/lib/python3.9/site-packages/thales_remote/script_wrapper.py", line 946, in setCVOutputFileName
    return self.setValue("CV_ROOT", name)
  File "/home/user/.local/lib/python3.9/site-packages/thales_remote/script_wrapper.py", line 1569, in setValue
    raise ThalesRemoteError(
thales_remote.error.ThalesRemoteError: ERROR;102;0
```

Any idea what `ERROR;102;0` could be? Also see #18

**Comment by maxkrapp1** [comment]
I can't reproduce this with this code.

```python
import sys
from thales_remote.connection import ThalesRemoteConnection
from thales_remote.script_wrapper import ThalesRemoteScriptWrapper

if __name__ == "__main__":
    zenniumConnection = ThalesRemoteConnection()
    zenniumConnection.connectToTerm("localhost", "ScriptRemote")

    zahnerZennium = ThalesRemoteScriptWrapper(zenniumConnection)
    zahnerZennium.forceThalesIntoRemoteScript()
    zahnerZennium.calibrateOffsets()

    zahnerZennium.setCVOutputPath(r"c:\\THALES")

    zahnerZennium.setCVOutputFileName("cv_series")
    zahnerZennium.setCVNaming("counter")
    zahnerZennium.setCVCounter(1)

    zahnerZennium.setCVStartPotential(1)
    zahnerZennium.setCVUpperReversingPotential(2)
    zahnerZennium.setCVLowerReversingPotential(0)
    zahnerZennium.setCVEndPotential(1)

    zahnerZennium.setCVStartHoldTime(2)
    zahnerZennium.setCVEndHoldTime(2)

    zahnerZennium.setCVCycles(1.5)
    zahnerZennium.setCVSamplesPerCycle(400)
    zahnerZennium.setCVScanRate(1)

    zahnerZennium.setCVMaximumCurrent(0.03)
    zahnerZennium.setCVMinimumCurrent(-0.03)

    zahnerZennium.setCVOhmicDrop(0)

    zahnerZennium.disableCVAutoRestartAtCurrentOverflow()
    zahnerZennium.disableCVAutoRestartAtCurrentUnderflow()
    zahnerZennium.disableCVAnalogFunctionGenerator()

    zahnerZennium.checkCVSetup()
    print(zahnerZennium.readCVSetup())

    zahnerZennium.measureCV()

    zenniumConnection.disconnectFromTerm()
    print("finish")
```

In general, the CV is somewhat special and complex with the errors. But this can not be improved in Thales.

**Comment by ilka-schulz** [comment]
I found the error: I used this string for the filename
```python
filename = str(datetime.datetime.now()) + " ees_lab_assistant CV"
device.setCVOutputFileName(filename)
```

This means there are two bugs on the Term software side:
1. The error message is wrong: `102` means `ERROR_ILLEGAL_COMMAND`. But it is actually an illegal parameter, not an illegal command.
2. The Term software sometimes goes into an unrecoverable state (see #13) and it is not possible to reconnect until one manually schedules a `Full Reset`. 

The Term software can also not handle colon (`:`) symbols in file names but I understand that this a Windows issue because Windows is extremely restricted in terms of file names. (https://stackoverflow.com/questions/1976007/what-characters-are-forbidden-in-windows-and-linux-directory-names) Linux spoiled me, I guess...

**Comment by maxkrapp1** [comment]
Yes I know the problem with the not allowed characters under windows.

I have added the term error to our internal ticket system.
At the end of the week is probably a new software with an optimized term software released, which should run more stable.

**Comment by maxkrapp1** [comment]
The situation is complicated in Remote2 this is not a bug.

filename = str(datetime.datetime.now()) + " ees_lab_assistant CV"

Then filename = "2023-02-13 10:35:24.119886 ees_lab_assistant CV".

In Remote2, however, ":" are the separators for commands. Illegal command then refers to the number 35 which is interpreted as a command.

**Comment by ilka-schulz** [comment]
I see... However, this basically means that the protocol is not able to handle strings, right? If you use a text-based protocol, you definitely want to encode strings in some way, e.g. url-encoded strings or base64 would be common encodings.

**Comment by maxkrapp1** [comment]
We have now added the characters allowed in file paths to the Remote2 manual.

https://doc.zahner.de/manuals/remote2.pdf page 5

These restrictions result on the one hand from the Windows conventions for paths, and on the other hand from the restrictions of the historically grown system, which only supports ASCII.
But I agree with you that in this historical protocol the mechanism to transfer strings was implemented later with the restrictions that exist now. An adaptation to modern string encodings as you suggested is not possible.
